Sources/Amplitude/Plugins/iOS/IOSLifecycleMonitor.swift

```diff
@@ -87,17 +90,74 @@
             }
         }
 
+        func applicationDidFinishLaunchingNotification(notification: NSNotification) {
+            amplitude?.apply { (ext) in
+                if let validExt = ext as? IOSLifecycle {
+                    validExt.applicationDidFinishLaunchingNotification(application: application)
+                }
+            }
+        }
     }
 
     extension AmplitudeDestinationPlugin: IOSLifecycle {
         public func applicationWillEnterForeground(application: UIApplication?) {
             let timestamp = Int64(NSDate().timeIntervalSince1970 * 1000)
             self.amplitude?.onEnterForeground(timestamp: timestamp)
+            if self.amplitude?.configuration.defaultTracking.appLifecycles == true {
+                let info = Bundle.main.infoDictionary
+                let currentBuild = info?["CFBundleVersion"] as? String
+                let currentVersion = info?["CFBundleShortVersionString"] as? String
+                self.amplitude?.track(eventType: Constants.AMP_APPLICATION_OPENED_EVENT, eventProperties: [
```

Comment:
> Two Application Opened events on app start - first with fromBackground == false (in didFinishLaunching callback) and second with fromBackground == true (in willEnterForeground callback)

> Two Application Opened events are triggered when I (re)start example app in simulator (XCode). If the app restored from background, only one Application Opened event is triggered.

I'm getting the same behavior (2 opened events).

I'm thinking in this case we should only fire app opened on `applicationDidFinishLaunchingNotification` (background false), since the app is being started "fresh" - not from background.

Maybe we can set a flag in `applicationDidFinishLaunchingNotification` and clear it in applicationDidEnterForeground/Background to prevent double firing.

Reply:
Fixed (added the flag). Removed unnecessary layers of interfaces/types and some unused property/code (`cleanup`).